File: factory/Geral/TesteApp.java
public class TesteApp {
    public static void main(String[] args) {
        Criador fabProduto1 = new CriadorConcreto1();
		Criador fabProduto2 = new CriadorConcreto2();


		Produto abc = fabProduto1.metodoFabrica();
        Produto def = fabProduto1.metodoFabrica();
        Produto ghi = fabProduto1.metodoFabrica();
        Produto jkl = fabProduto2.metodoFabrica();

    }
}

File: factory/Geral/src/TesteApp.ts
class TesteApp {
    public static main(): void {
        const fabProduto1: Criador = new CriadorConcreto1();
        const fabProduto2: Criador = new CriadorConcreto2();

        const abc: Produto = fabProduto1.metodoFabrica();
        const def: Produto = fabProduto1.metodoFabrica();
        const ghi: Produto = fabProduto1.metodoFabrica();
        const jkl: Produto = fabProduto2.metodoFabrica();
    }
}

// Chama o método main para executar o código
TesteApp.main();
